#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_gtk-immodule-install-dir.dpatch by Ming Hua <minghua@rice.edu>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Change the gtk immodule installation directory from
## DP: /usr/lib/gtk-2.0/immodules/ to /usr/lib/gtk-2.0/2.4.0/immodules/.
## DP:
## DP: This is just a workaround for the problem that
## DP: /usr/sbin/update-gtk-immodules used by GTK+ packages only looks
## DP: for the specific versioned directory (so that 2.2.x and 2.4.x can
## DP: coexist, probably).  I can't use /usr/bin/gtk-query-immodules-2.0
## DP: instead because it completely rewrites /etc/gtk-2.0/gtk.immodules
## DP: (therefore break other packages), and the change won't survive
## DP: upgrades of libgtk2.0-0 and other input method packages that use
## DP: /usr/sbin/update-gtk-immodules.
## DP:
## DP: This patch only changes extras/gtk_immodule/Makefile.am, and the
## DP: change necessary for extras/gtk_immodule/Makefile.in is
## DP: incorporated in 11_relibtoolize.dpatch.  If the relibtoolize
## DP: patch is removed, this patch should change Makefile.in as well.
## DP:
## DP: Note that this patch will break for GTK+ 2.10.x as there is an
## DP: ABI change for the immodules, and the installation directory
## DP: should be changed to /usr/lib/gtk-2.0/2.10.0/immodules/ when
## DP: built against GTK+ 2.10.x.

@DPATCH@
diff -urNad scim-1.4.4~/extras/gtk2_immodule/Makefile.am scim-1.4.4/extras/gtk2_immodule/Makefile.am
--- scim-1.4.4~/extras/gtk2_immodule/Makefile.am	2005-07-10 08:45:36.000000000 -0500
+++ scim-1.4.4/extras/gtk2_immodule/Makefile.am	2006-09-10 23:22:07.000000000 -0500
@@ -32,7 +32,7 @@
 
 noinst_HEADERS = gtkimcontextscim.h
 
-moduledir = @GTK_LIBDIR@/gtk-2.0/immodules
+moduledir = @GTK_LIBDIR@/gtk-2.0/2.4.0/immodules
 
 module_LTLIBRARIES = $(CONFIG_GTK2_IMMODULE)
 
